app/components/request/configs.js

```diff
@@ -16,63 +16,106 @@ export default Component.extend(CanTriggerBuild, TriggerBuild, {
   classNames: ['request-configs'],
   classNameBindings: ['status'],
 
-  requestConfig: service(),
+  preview: service('request-config'),
   features: service(),
-
-  status: 'closed',
-  closed: equal('status', 'closed'),
-  customizing: equal('status', 'customize'),
-  previewing: equal('status', 'preview'),
-  replacing: equal('mergeMode', 'replace'),
   showNewConfigView: reads('features.showNewConfigView'),
 
   repo: reads('request.repo'),
-  rawConfigs: or('requestConfig.rawConfigs', 'request.uniqRawConfigs'),
-  messages: reads('requestConfig.messages'),
-  loading: reads('requestConfig.loading'),
+  repoId: reads('repo.id'),
+  rawConfigs: or('preview.rawConfigs', 'request.uniqRawConfigs'),
+  messages: reads('preview.messages'),
+  loading: reads('preview.loading'),
 
   refType: 'sha',
-  sha: truncated('originalSha', 10),
+  sha: reads('originalSha'),
   branch: reads('originalBranch'),
   message: reads('request.commit.message'),
   config: reads('request.apiConfig.config'),
   mergeMode: reads('originalMergeMode'), // TODO store and serve merge mode for api request configs
   defaultMergeMode: 'deep_merge_append',
 
-  originalSha: or('requestSha', 'repoDefaultBranchLastCommitSha'),
+  originalSha: truncated('requestOrDefaultBranchSha', 10),
   originalBranch: or('requestBranch', 'repoDefaultBranch'),
   originalMergeMode: or('request.mergeMode', 'defaultMergeMode'),
   requestBranch: reads('request.branchName'),
   requestSha: reads('request.commit.sha'),
+  requestOrDefaultBranchSha: or('requestSha', 'repoDefaultBranchLastCommitSha'),
   repoDefaultBranch: reads('repo.defaultBranch.name'),
   repoDefaultBranchLastCommitSha: reads('repo.defaultBranch.lastBuild.commit.sha'),
 
-  onTriggerBuild(e) {
+  ref: computed('refType', 'branch', 'sha', function () {
+    return this.get(this.refType);
+  }),
+
+  didInsertElement() {
+    this.load();
+  },
+
+  onTrigger(e) {
     e.toElement.blur();
-    if (this.status == 'closed') {
+    if (this.closed) {
       this.set('status', 'open');
-    } else if (['open', 'customize', 'preview'].includes(this.status)) {
+    } else {
       this.submit();
     }
   },
 
   onCustomize() {
-    if (this.status == 'customize') {
+    if (this.customizing) {
       this.set('status', 'open');
       this.reset();
     } else {
       this.set('status', 'customize');
     }
   },
 
+  onPreview() {
+    if (!this.previewing) {
+      this.set('status', 'preview');
+      if (!this.loaded) {
+        this.load();
+      }
+    } else if (this.customized) {
```

Comment:
no, it is going to be set if the user has changed any of the form values (i.e. "customized" some of the values on the "customize" form). 

this is one of the bigger UX questions of how the buttons should behave. i think it will be best to have another discussion about this once we have the final branch done, and can look at a branch deployment together.

let's keep this for now.

Reply:
Sounds reasonable, thank you for explaining 👍